src/Workspaces/Core/Portable/Classification/ClassifierHelper.cs

```diff
@@ -189,14 +208,20 @@ private static void AdjustSpans(ArrayBuilder<ClassifiedSpan> spans, TextSpan wid
 
                 if (replacementIndex >= 0)
                 {
+                    // There may be multiple semantic parts corresponding to a single
+                    // syntactic part, so we might need to go through a syntactic part
+                    // multiple times to verify. For example, this is the case with
+                    // verbatim string literals containing string escape characters.
+                    i--;
```

Comment:
thanks for the example.  do we have ests for this?

Reply:
Yep, a test in `SemanticTokensRangeTests` actually caught this error 😄

Reply:
ok great!

Reply:
actually, can you ensure we have a few tests here.  for example, if a verbatim literal contains multiple escape characters like so:

`$"foo""bar""baz{x + y}a""b""c"`

something to stress teh system

Reply:
I added a regex test in my latest push which I believe accomplishes this 🙂